projects/solar/test/test_relay_fsm.c

```diff
@@ -23,35 +24,62 @@
 
 #define IRRELEVANT_EVENT NUM_SOLAR_RELAY_EVENTS
 
-#define PROCESS_RELAY_FSM_EVENT(storage, e)    \
-  ({                                           \
-    MS_TEST_HELPER_AWAIT_EVENT(e);             \
-    relay_fsm_process_event(&(storage), &(e)); \
-    MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();   \
+#define PROCESS_RELAY_FSM_OPEN_EVENT(storage, e)                      \
+  ({                                                                  \
+    MS_TEST_HELPER_AWAIT_EVENT(e);                                    \
+    relay_fsm_process_event(&(storage), &(e));                        \
+    MS_TEST_HELPER_CAN_TX_RX(SOLAR_CAN_EVENT_TX, SOLAR_CAN_EVENT_RX); \
+    TEST_ASSERT_TRUE(relay_6_mppts_rx_cb_called);                     \
+    MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();                          \
+  })
+#define PROCESS_RELAY_FSM_OTHER_EVENT(storage, e) \
+  ({                                              \
+    MS_TEST_HELPER_AWAIT_EVENT(e);                \
+    relay_fsm_process_event(&(storage), &(e));    \
+    MS_TEST_HELPER_ASSERT_NO_EVENT_RAISED();      \
   })
-
 static CanStorage s_can_storage;
 static RelayFsmStorage s_storage;
 
-static bool relay_err_rx_cb_called;
-static StatusCode prv_err_rx_cb(const CanMessage *msg, void *context, CanAckStatus *ack_reply) {
-  relay_err_rx_cb_called = true;
+static bool relay_error_rx_cb_called;
+static bool relay_5_mppts_rx_cb_called;
+static bool relay_6_mppts_rx_cb_called;
+static uint8_t error_array[5];
+static int current_error_num;
+
+static StatusCode prv_error_rx_cb(const CanMessage *msg, void *context, CanAckStatus *ack_reply) {
+  relay_error_rx_cb_called = true;
   uint8_t solar_fault;
   uint8_t fault_data;
   CAN_UNPACK_SOLAR_FAULT_6_MPPTS(msg, &solar_fault, &fault_data);
-  TEST_ASSERT_EQUAL(EE_SOLAR_FAULT_DRV120, solar_fault);
-  TEST_ASSERT_EQUAL(0, fault_data);
+  error_array[current_error_num] = solar_fault;
+  current_error_num++;
+  return STATUS_CODE_OK;
+}
+static StatusCode prv_current_5_mppts_rx_cb(const CanMessage *msg, void *context,
+                                            CanAckStatus *ack_reply) {
+  relay_5_mppts_rx_cb_called = true;
+  return STATUS_CODE_OK;
+}
+static StatusCode prv_current_6_mppts_rx_cb(const CanMessage *msg, void *context,
```

Comment:
Nit: there should be blank lines between the functions here.

Reply:
done